### PR TITLE
docs: update documentation for `ClientStream.SendMsg()` returning `nil` unconditionally when `ClientStreams=false`

### PR DESCRIPTION
**Pull request by hanut19**
Addresses: #7230 

RELEASE NOTES: None

### CODE REVIEW

**Review by purnesh42H — APPROVED** [review]
lgtm

File: stream.go [hunk 1]
```diff
@@ -113,7 +113,9 @@ type ClientStream interface {
 	// SendMsg is generally called by generated code. On error, SendMsg aborts
 	// the stream. If the error was generated by the client, the status is
 	// returned directly; otherwise, io.EOF is returned and the status of
-	// the stream may be discovered using RecvMsg.
+	// the stream may be discovered using RecvMsg. However, if
+	// StreamDesc.ClientStreams was set to false, then nil error is returned
```
Comment (purnesh42H) [inline stream.go:117]: nit: `StreamDesc.ClientStreams`

Comment (dfawley) [reply stream.go:117]: Sorry, what is being suggested here?

Comment (dfawley) [inline stream.go:117]: How about this instead for the last sentence:

```
For unary or server-streaming RPCs (StreamDesc.ClientStreams is false),
a nil error is returned unconditionally.
```

Comment (hanut19) [reply stream.go]: done please review

**Review by dfawley — APPROVED** [review]

**Pull request merged**